### PR TITLE
shortage: Get the correct item name for the missing entity name

### PR DESCRIPTION
**Pull request by bengardner**
Known to correct:
  stone-path => stone-brick
  curved-rail => rail x4
  straight-rail => rail
  cliff => cliff-explosives

### CODE REVIEW

File: src/GlobalState.lua [hunk 1]
```diff
@@ -510,4 +510,29 @@ function M.update_queue(update_entity)
   Queue.swap_random_to_front(global.mod.scan_queue, global.mod.rand)
 end
 
+-- translate a tile name to the item name ("stone-path" => "stone-brick")
+function M.resolve_name(name)
+  if game.item_prototypes[name] ~= nil or game.fluid_prototypes[name] ~= nil then
```
Comment (year6b7a) [inline src/GlobalState.lua:515]: In what situtation would a fluid be requested in a construction alert?

File: src/GlobalState.lua [hunk 2]
```diff
@@ -510,4 +510,29 @@ function M.update_queue(update_entity)
...
+
+  local prot = game.tile_prototypes[name]
+  if prot ~= nil then
+    local mp = prot.mineable_properties
+    if mp.minable and #mp.products == 1 then
+      return mp.products[1].name
+    end
+  end
+
+  -- FIXME: figure out how to not hard-code this
```
Comment (year6b7a) [inline src/GlobalState.lua:527]: Do you have any ideas for how to not hard-code this information?

**Review by year6b7a — COMMENTED** [review]
This looks good to me. I have a few questions but will merge soon.

File: src/GlobalState.lua (hunk 1 above)
Comment (bengardner) [reply src/GlobalState.lua:515]: A fluid would never be in an alert. I had intended this function to be a generic "map some name to a valid thing we manage", but then it became specific to the alerts.
The fluid check could be removed and the function could be moved to where it is used.

File: src/GlobalState.lua (hunk 2 above)
Comment (bengardner) [reply src/GlobalState.lua:527]: Nope. I haven't found "curved-rail" in a runtime prototype structure.
It is at data.raw["curved-rail"]["curved-rail"] in the data stage. Not promising.

**Pull request merged**